src/app/qgsattributetabledialog.cpp

```diff
@@ -785,25 +784,33 @@ void QgsAttributeTableDialog::mActionCopySelectedRows_triggered()
       }
       fieldNames << columnConfig.name;
     }
-    featureStore.setFields( fields );
-
-    QgsFeatureIterator it = mLayer->getFeatures( QgsFeatureRequest( qgis::listToSet( featureIds ) )
-                            .setSubsetOfAttributes( fieldNames, mLayer->fields() ) );
-    QgsFeatureMap featureMap;
-    QgsFeature feature;
-    while ( it.nextFeature( feature ) )
+    if ( fields != mLayer->fields() )
     {
-      QgsVectorLayerUtils::matchAttributesToFields( feature, fields );
-      featureMap[feature.id()] = feature;
+      QgsFeatureStore featureStore;
+      featureStore.setFields( fields );
+
+      QgsFeatureIterator it = mLayer->getFeatures( QgsFeatureRequest( qgis::listToSet( featureIds ) )
+                              .setSubsetOfAttributes( fieldNames, mLayer->fields() ) );
+      QgsFeatureMap featureMap;
+      QgsFeature feature;
+      while ( it.nextFeature( feature ) )
+      {
+        QgsVectorLayerUtils::matchAttributesToFields( feature, fields );
+        featureMap[feature.id()] = feature;
+      }
+      for ( const QgsFeatureId &id : featureIds )
+      {
+        featureStore.addFeature( featureMap[id] );
+      }
+
+      featureStore.setCrs( mLayer->crs() );
+
+      QgisApp::instance()->clipboard()->replaceWithCopyOf( featureStore );
     }
-    for ( const QgsFeatureId &id : featureIds )
+    else
     {
-      featureStore.addFeature( featureMap[id] );
+      QgisApp::instance()->copySelectionToClipboard( mLayer );
```

Comment:
This isn't equivalent -- the featureIds list will be only the selected rows which are visible in the view, ie. excluding those hidden by a filter on the table

Reply:
@nyalldawson , Interesting, I hadn't realized that. Are we not hitting the same issue with dual view mode then?